x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -412,6 +417,25 @@ public String toString() {
             return name;
         }
 
+        private static List<String> parseClaimValues(JWTClaimsSet claimsSet, String claimName, String settingKey) {
+            List<String> values;
+            final Object claimValueObject = claimsSet.getClaim(claimName);
+            if (claimValueObject == null) {
+                values = List.of();
+            } else if (claimValueObject instanceof String) {
+                values = List.of((String) claimValueObject);
+            } else if (claimValueObject instanceof Collection) {
+                if (claimValueObject instanceof Collection && ((Collection) claimValueObject).stream().allMatch(c -> c instanceof String)) {
+                    values = (List<String>) claimValueObject;
+                } else {
+                    throw new SettingsException("Setting [ " + settingKey + " expects a claim with String or a String Array value");
+                }
+            } else {
+                throw new SettingsException("Setting [ " + settingKey + " expects a claim with String or a String Array value");
+            }
```

Comment:
* The check `claimValueObject instanceof Collection` is performed twice, which does not seem to be necessary
* The `if` check only ensure it is a `Collection<String>`, not a `List<String>`. So there is a theoretical possibility for cast exception.

I feel this part of code can be simplified to something like the follows:
```java
} else if (claimValueObject instanceof Collection && ((Collection) claimValueObject).stream().allMatch(c -> c instanceof String)) {
    values = (Collection<String>) claimValueObject;
} else {
    throw new SettingsException("Setting [ " + settingKey + " expects a claim with String or a String Array value");
}
```

With above change, the return type of this method needs to be changed to `Collection<String>` as well.

Reply:
Yeap you are right. In retrospect, I should have waited to fix this today and not last night. Thanks for calling me out !